src/main/java/inventorysetups/InventorySetupsPersistentDataManager.java

```diff
@@ -108,14 +106,30 @@ public void updateConfig(boolean updateSetups, boolean updateSections)
 	{
 		if (updateSetups)
 		{
-			List<InventorySetupSerializable> issList = new ArrayList<>();
+			// Rather than escaping the name to pick a config key, instead use hash of the name. Benefits:
+			//   Changes to the escaping function or supported characters in the name string don't affect config keys.
+			//   Standardizes length; preventing overflow issues from extremely long names
+			//   Keeps json as the single source of truth for names; other plugins won't try to grab names from config keys.
+			final String wholePrefix = ConfigManager.getWholeKey(CONFIG_GROUP, null, CONFIG_KEY_SETUPS_V3_PREFIX);

```

Comment:
can you explain what the null is? I see it's profile. Is that rs profile as in rs account or runelite profile?

Reply:
It looks like there is now support for saving directly to specific configuration profiles; but the old pattern of not specifying a profile remains intact. In discord conversation with Adam, he made no suggestion of switching to the new pattern, so it seems that what this plugin currently does (not specify profile) will remain supported in the future.

Reply:
> I believe it is rsProfile; there are some `getWholeKey(groupName, rsProfile, key)` calls in the same file. that `getWholeKey` is defined in.
> 
> It looks like there is support for saving directly to specific rsProfile configurations; but the current pattern of not specifying an rsProfile remains intact. In discord conversation with Adam, he made no suggestion of switching to the new pattern, so it seems that what this plugin currently does (not specify rsProfile) will remain supported in the future.